### PR TITLE
Configurable persistence of friendly name to KV

### LINKED ISSUE #7562: Allow storing of `friendly_name` in KV when integrating with external auth API
When integrating with an external Auth API, the `friendly_name_claim_name` config parameter can be set, which extracts a claim value and sets it as the logged-in user's `friendly_name` attribute.  
However, this value doesn't persist in the user's KV store record. This means that it's visible to the logged-in user and can be used to show the friendly name in commits, etc, but when listing users, the ID is displayed in place of the `friendly_name` because it's not persisted to the KV store.

This sacrifices usability and administration capabilities for privacy. We want to make this configurable so that users can choose based on their specific administration needs, privacy/security situation, and requirements.

`friendly_name`s will not be stored by default for backward compatibility.

### PR DESCRIPTION
**Pull request by eladlachmi**
Closes #7562 

### Background

Please see the linked issue for context.

### New Feature

This PR adds two new configuration flags with a default value of `false`:

1. `cookie_auth_verification.persist_friendly_name` - SAML-based auth API
2. `oidc.persist_friendly_name` - OIDC-based auth API

When this flag is set to `true`, the friendly name claim pointed to by `friendly_name_claim_name` is saved with the user's record in the KV store. Each of the flags is only relevant to its respective authentication method.

Before:
Screenshot 2024-03-18 at 13 06 17

After:
after-screenshot
**Note: domain prefix intentionally removed.**

### Testing Details

Both positive and negative tests were done manually.

### Breaking Change?

None.

### CODE REVIEW

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]

File: docs/reference/configuration.md [hunk 1]
```diff
@@ -101,11 +101,13 @@ This reference uses `.` to denote the nesting of values.
 * `auth.cookie_auth_verification.default_initial_groups` (string[] : [])` - By default, users will be assigned to these groups
 * `auth.cookie_auth_verification.initial_groups_claim_name` `(string[] : [])` - Use this claim from the ID token to provide the initial group for new users. This will take priority if `auth.cookie_auth_verification.default_initial_groups` is also set.
 * `auth.cookie_auth_verification.friendly_name_claim_name` `(string[] : )` - If specified, the value from the claim with this name will be used as the user's display name.
+* `auth.cookie_auth_verification.persist_friendly_name` `(string : false)` - If set to `true`, the friendly name is persisted to the KV store and can be displayed in the user list. This is meant to be used in conjunction with `auth.oidc.friendly_name_claim_name`.
```
Comment (Isan-Rivkin) [inline docs/reference/configuration.md:104]:
```suggestion
* `auth.cookie_auth_verification.persist_friendly_name` `(string : false)` - If set to `true`, the friendly name is persisted to the KV store and can be displayed in the user list. This is meant to be used in conjunction with `auth.cookie_auth_verification.friendly_name_claim_name`.
```

File: docs/reference/security/authentication.md [hunk 2]
```diff
@@ -78,7 +78,8 @@ auth:
     callback_base_url: https://lakefs.example.com       # The scheme, domain (and port) of your lakeFS installation
     url: https://my-account.oidc-provider-example.com
     default_initial_groups: ["Developers"]
-    friendly_name_claim_name: name                      #  Optional: use the value from this claim as the user's display name 
+    friendly_name_claim_name: name                      #  Optional: use the value from this claim as the user's display name
+    persist_friendly_name: true                         #  Optional: persist friendly name to KV store so it can be displayed in the user list
```
Comment (Isan-Rivkin) [inline docs/reference/security/authentication.md:82]: Nice! please add a small section here about this field explaining the persistence lifecycle? (i.e when is it being updated etc)

File: pkg/api/auth_middleware.go [hunk 3]
```diff
@@ -191,9 +194,15 @@ func checkSecurityRequirements(r *http.Request,
...
 		user.FriendlyName = &friendlyName
+		if persistFriendlyName && swag.StringValue(user.FriendlyName) != friendlyName {
+			err := authService.UpdateUserFriendlyName(ctx, user.Username, friendlyName)
+			if err != nil {
+				logger.WithError(err).Error("update user friendly name")
+			}
+		}
 	}
 	return user
 }
```
Comment (Isan-Rivkin) [inline pkg/api/auth_middleware.go:208]:
```suggestion
func enhanceWithFriendlyName(ctx context.Context, user *model.User, friendlyName string, persistFriendlyName bool, authService auth.Service, logger logging.Logger) *model.User {
	if swag.StringValue(user.FriendlyName) != friendlyName {
		user.FriendlyName = swag.String(friendlyName)
		if persistFriendlyName {
			if err := authService.UpdateUserFriendlyName(ctx, user.Username, friendlyName); err != nil {
				logger.WithError(err).Error("failed to update user friendly name")
			}
		}

	}
	return user
}
```

**Review by Isan-Rivkin — CHANGES_REQUESTED** [review]
minor comments :)

File: docs/reference/configuration.md (hunk 1 above)
Comment (eladlachmi) [reply docs/reference/configuration.md]: Fixed

File: docs/reference/security/authentication.md (hunk 2 above)
Comment (eladlachmi) [reply docs/reference/security/authentication.md:82]: Added a section. Let me know what you think.

File: pkg/api/auth_middleware.go (hunk 3 above)
Comment (eladlachmi) [reply pkg/api/auth_middleware.go]: Done

**Review by Isan-Rivkin — APPROVED** [review]
LGTM!

**Pull request merged**